game-core/src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -438,22 +439,24 @@ private void runStep(final boolean stepIsRestoredFromSavedGame) {
     if (gameData.getSequence().next()) {
       gameData.getHistory().getHistoryWriter().startNextRound(gameData.getSequence().getRound());
       saveGame(gameData.getSequence().getRound() % 2 == 0
-          ? SaveGameFileChooser.getEvenRoundAutoSaveFile(HeadlessGameServer.headless())
-          : SaveGameFileChooser.getOddRoundAutoSaveFile(HeadlessGameServer.headless()));
+          ? SaveGameFileChooser.getEvenRoundAutoSaveFile(headless)
+          : SaveGameFileChooser.getOddRoundAutoSaveFile(headless));
     }
     if (autoSaveThisDelegate && !currentStep.getName().endsWith("Move")) {
-      autoSaveAfter(currentDelegate);
+      autoSaveAfter(currentDelegate, headless);
     }
   }
 
-  private void autoSaveAfter(final GameStep gameStep) {
-    saveGame(SaveGameFileChooser.getAfterStepAutoSaveFile(gameStep.getName()));
+  private void autoSaveAfter(final String stepName, final boolean headless) {
+    saveGame(SaveGameFileChooser.getAfterStepAutoSaveFile(
+        headless ? (stepName.endsWith("NonCombatMove") ? "NonCombatMove" : "CombatMove") : stepName, headless));
```

Comment:
I saw you mention someplace about having to remove the player name from the step name when headless.  Is that what this expression is doing?  If so, to avoid someone scratching their head in the future :smile:, might I suggest either a comment stating this, or even an intermediate variable whose name describes this operation?

Also, since you are asserting that this is a step whose name ends with `Move` in the caller (line 432), would it make sense to move this logic up there so all of it is co-located?  Something like:

```java
if (autoSaveThisDelegate && currentStep.getName().endsWith("Move")) {
  final String stepName = headless
      // remove player name from step name because XXX
      ? (currentStep.getName().endsWith("NonCombatMove") ? "NonCombatMove" : "CombatMove")
      : currentStep.getName();
  autoSaveAfter(stepName, headless);
}
```